`native-maven-plugin` cannot publish to local repository on Windows with Simplified Chinese

**Before reporting**

- This repository should be used to report **issues on the Maven or Gradle plugins for GraalVM**.
- Please report issues which are specific to [the Spring Framework](https://spring.io/) or [the Micronaut framework](https://micronaut.io/) to their specific repositories.
- Do not report issues with building your specific application, e.g errors which happen at image build time like classes initialized at build time, or missing classes as run time: those are not related to the plugins but problems with configuration. You can refer to the [GraalVM native image documentation](https://www.graalvm.org/latest/reference-manual/native-image/) for available options and the [plugins documentation](https://graalvm.github.io/native-build-tools) for how to use them with the plugin.

**Describe the bug**
A clear and concise description of what the bug is.
**Make sure that you have read [the documentation](https://graalvm.github.io/native-build-tools) and that you are using the latest plugin version.**

- `native-maven-plugin` cannot publish to local repository on Windows 11 Home 24H2 with Simplified Chinese.

**To Reproduce**

When possible, provide a link to a repository which reproduces the issue, with instructions on how to use.
The reproducer **must** make use of either the Maven or Gradle plugin.

Steps to reproduce the behavior:

- Execute the following PowerShell 7 command on the `Windows 11 Home 24H2` instance with `PowerShell/PowerShell`,
  `version-fox/vfox`, `git-for-windows/git`, `rancher-sandbox/rancher-desktop` and
  `Microsoft.VisualStudio.2022.Community` installed.

```shell
vfox add java
vfox install java@17.0.9-graalce
vfox use --global java@17.0.9-graalce

git clone git@github.com:graalvm/native-build-tools.git
cd ./native-build-tools/
./gradlew publishToMavenLocal --no-parallel
```


Please use backticks to [properly format code](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting).
If possible please attach a complete reproducer here (either as [a zip file](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) or as a link to public repository/branch).

**Expected behavior**
A clear and concise description of what you expected to happen.

- `native-maven-plugin` can publish to local repository on Windows with Simplified Chinese.

**Logs**
Add logs to help explain your problem. 
Please use backticks to [properly format big logs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks). Example:

<details>

<summary>Click me to see the full Log🛝🎟️🫠🛝🛝🎑😊😁🛝🎟️🎑🎑🎋🫠</summary>

```shell
PS D:\TwinklingLiftWorks\git\public\native-build-tools> ./gradlew publishToMavenLocal --no-parallel

> Configure project :native-gradle-plugin
Test task functionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle7.4FunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle7.6.2FunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.0.1FunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.2.1FunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.13FunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task configCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle7.4ConfigCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle7.6.2ConfigCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.0.1ConfigCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.2.1ConfigCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java
Test task gradle8.13ConfigCacheFunctionalTest will use GRAALVM_HOME = C:\Users\lingh\.version-fox\temp\1748707200-23404\java

> Task :native-maven-plugin:javadoc FAILED
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:259: 错误: 编码 GBK映射字符 (0x80)
     * 鈹溾攢鈹? apache/
           ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:260: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹溾攢鈹? commons/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:260: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹溾攢鈹? commons/
                ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:261: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹溾攢鈹? validator/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:261: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹溾攢鈹? validator/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:261: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹溾攢鈹? validator/
                     ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:262: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? UrlValidator.class
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:262: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? UrlValidator.class
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:262: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? UrlValidator.class
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:262: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? UrlValidator.class
                          ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:263: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? routines/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:263: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? routines/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:263: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? routines/
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:263: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? routines/
                          ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:264: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? ValidatorUtils.class
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:264: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? ValidatorUtils.class
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:264: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? ValidatorUtils.class
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:264: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? ValidatorUtils.class
                       ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:264: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? ValidatorUtils.class
                               ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:265: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? util/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:265: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? util/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:265: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? util/
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:265: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? util/
                          ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:266: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? Flags.class
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:266: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? Flags.class
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:266: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? Flags.class
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:266: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? Flags.class
                       ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:266: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? Flags.class
                               ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:267: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹溾攢鈹? digester/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:267: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹溾攢鈹? digester/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:267: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹溾攢鈹? digester/
                     ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:268: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? plugins/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:268: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? plugins/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:268: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? plugins/
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:268: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹溾攢鈹? plugins/
                          ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:269: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? strategies/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:269: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? strategies/
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:269: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? strategies/
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:269: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? strategies/
                       ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:269: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹?   鈹斺攢鈹? strategies/
                               ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:270: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?       鈹斺攢鈹? DigesterPlugin.class
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:270: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?       鈹斺攢鈹? DigesterPlugin.class
             ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:270: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?       鈹斺攢鈹? DigesterPlugin.class
                  ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:270: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹?   鈹?   鈹?       鈹斺攢鈹? DigesterPlugin.class
                       ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:270: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹?   鈹?   鈹?       鈹斺攢鈹? DigesterPlugin.class
                                   ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:271: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹溾攢鈹? json/
           ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:272: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?   鈹斺攢鈹? org/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:272: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?   鈹斺攢鈹? org/
                ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:273: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?       鈹斺攢鈹? json/
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:273: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?       鈹斺攢鈹? json/
                    ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:274: 错误: 编码 GBK 的不可映射字符 (0x82)
     * 鈹?           鈹斺攢鈹? JSONObject.class
        ^
D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\src\main\java\org\graalvm\buildtools\maven\sbom\ArtifactAdapterResolver.java:274: 错误: 编码 GBK 的不可映射字符 (0x80)
     * 鈹?           鈹斺攢鈹? JSONObject.class
                        ^
52 个错误

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':native-maven-plugin:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): 'D:\TwinklingLiftWorks\git\public\native-build-tools\native-maven-plugin\build\tmp\javadoc\javadoc.options'

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
126 actionable tasks: 18 executed, 108 up-to-date
```

</details>

**System Info (please complete the following information):**
 - OS: [e.g. `Windows`] `Windows`
 - GraalVM Version [e.g. `22.0 CE`] `CE For JDK 17.0.9`
 - Java Version [e.g. `17`] `17`
 - Plugin version [e.g. `native-gradle-plugin:0.9.10`] `native-gradle-plugin:0.11.0-SNAPSHOT`

**Additional context**
Add any other context about the problem here.

- Just add the following to the `native-maven-plugin/build.gradle.kts` file to fix it.
```kotlin
tasks {
    withType<Javadoc>().configureEach { options.encoding = "UTF-8" }
}
```

--- a/docs/src/docs/asciidoc/changelog.adoc
+++ b/docs/src/docs/asciidoc/changelog.adoc
@@ -12,6 +12,7 @@ This version introduces a breaking change: the plugin now requires Java 17 to ru
 === Maven plugin
 
 - Added support for running integration tests via maven-failsafe-plugin
+- Add `runtimeArgs` support to `native-maven-plugin`
 
 == Release 0.10.6
 

--- a/docs/src/docs/asciidoc/maven-plugin.adoc
+++ b/docs/src/docs/asciidoc/maven-plugin.adoc
@@ -119,6 +119,17 @@ The following configuration options are available:
     <propertyName>value</propertyName>
 </systemPropertyVariables>
 ----
+`<runtimeArgs>`::
+To specify runtime arguments used for a native image run, use:
+[source,xml, role="multi-language-sample"]
+----
+<runtimeArgs>
+    <runtimeArg>-XX:MissingRegistrationReportingMode=Warn</runtimeArg>
+</runtimeArgs>
+----
+
+This parameter is only valid for the Maven Goal of `test`.
+
 `<jvmArgs>`::
    To specify JVM arguments used for a native image build, use:
 [source,xml, role="multi-language-sample"]

--- a/native-maven-plugin/build.gradle.kts
+++ b/native-maven-plugin/build.gradle.kts
@@ -178,3 +178,7 @@ tasks.withType<Checkstyle>().configureEach {
     // generated code
     exclude("**/RuntimeMetadata*")
 }
+
+tasks {
+    withType<Javadoc>().configureEach { options.encoding = "UTF-8" }
+}

--- a/native-maven-plugin/src/functionalTest/groovy/org/graalvm/buildtools/maven/JavaApplicationWithTestsFunctionalTest.groovy
+++ b/native-maven-plugin/src/functionalTest/groovy/org/graalvm/buildtools/maven/JavaApplicationWithTestsFunctionalTest.groovy
@@ -41,8 +41,10 @@
 
 package org.graalvm.buildtools.maven
 
+import org.graalvm.buildtools.utils.NativeImageUtils
 import spock.lang.IgnoreIf
 import spock.lang.Issue
+import spock.lang.Requires
 
 class JavaApplicationWithTestsFunctionalTest extends AbstractGraalVMMavenFunctionalTest {
 
@@ -182,4 +184,21 @@ class JavaApplicationWithTestsFunctionalTest extends AbstractGraalVMMavenFunctio
         outputDoesNotContain expectedOutput
     }
 
+    private static int getCurrentJDKVersion() {
+        return NativeImageUtils.getMajorJDKVersion(GraalVMSupport.getGraalVMHomeVersionString())
+    }
+
+    @Requires({ getCurrentJDKVersion() >= 23 })
+    def "can use the Maven plugin with the runtimeArgs config to run tests in a native image"() {
+        withSample("java-application-with-tests")
+
+        when:
+        mvn '-Ptest-runtime-args', '-DquickBuild', 'test'
+
+        def expectedOutput = "Note: this run will print partial stack traces of the locations where a"
+
+        then:
+        buildSucceeded
+        outputContains expectedOutput
+    }
 }

--- a/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java
+++ b/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java
@@ -168,6 +168,9 @@ public abstract class AbstractNativeImageMojo extends AbstractNativeMojo {
     @Parameter(property = "jvmArgs")
     protected List<String> jvmArgs;
 
+    @Parameter(property = "runtimeArgs")
+    protected List<String> runtimeArgs;
+
     @Parameter(property = NATIVE_IMAGE_DRY_RUN, defaultValue = "false")
     protected boolean dryRun;
 

--- a/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeTestMojo.java
+++ b/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeTestMojo.java
@@ -168,6 +168,9 @@ public class NativeTestMojo extends AbstractNativeImageMojo {
         }
         systemProperties.put("junit.platform.listeners.uid.tracking.output.dir",
             NativeExtension.testIdsDirectory(outputDirectory.getAbsolutePath()));
+        if (runtimeArgs == null) {
+            runtimeArgs = new ArrayList<>();
+        }
 
         imageName = NATIVE_TESTS_EXE;
         mainClass = "org.graalvm.junit.platform.NativeImageJUnitLauncher";
@@ -238,6 +241,7 @@ public class NativeTestMojo extends AbstractNativeImageMojo {
             command.add("--xml-output-dir");
             command.add(xmlLocation.toString());
             systemProperties.forEach((key, value) -> command.add("-D" + key + "=" + value));
+            command.addAll(runtimeArgs);
 
             processBuilder.command().addAll(command);
             processBuilder.environment().putAll(environment);

--- a/samples/java-application-with-tests/pom.xml
+++ b/samples/java-application-with-tests/pom.xml
@@ -238,6 +238,41 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>test-runtime-args</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>3.0.0-M5</version>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.graalvm.buildtools</groupId>
+                        <artifactId>native-maven-plugin</artifactId>
+                        <version>${native.maven.plugin.version}</version>
+                        <extensions>true</extensions>
+                        <executions>
+                            <execution>
+                                <id>test-native</id>
+                                <goals>
+                                    <goal>test</goal>
+                                </goals>
+                                <phase>test</phase>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <buildArgs>
+                                <buildArg>--exact-reachability-metadata</buildArg>
+                            </buildArgs>
+                            <runtimeArgs>
+                                <runtimeArg>-XX:MissingRegistrationReportingMode=Warn</runtimeArg>
+                            </runtimeArgs>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
     </profiles>
 
     <build>

